cmd/server/main.go

```diff
@@ -39,80 +38,64 @@ type Arguments struct {
 
 func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) error {
 	var argsParsed Arguments
+
 	if err := utils.ParseFlags(args, &argsParsed); err != nil {
 		return err
 	}
+
 	if argsParsed.Port == 0 {
+		golog.Global.Debugf("using default port %d ", defaultPort)
 		argsParsed.Port = utils.NetPortFlag(defaultPort)
 	}
 
-	lidarComponents := search.Devices()
-	if len(lidarComponents) != 0 {
-		golog.Global.Debugf("detected %d lidar devices", len(lidarComponents))
-		for _, comp := range lidarComponents {
+	// Check if USB device is available
+	// TODO(jeremy): add search filter for product and model info to confirm rplidar is present instead of assuming
```

Comment:
This TODO is either already done it looks like (so comment needs to be removed) or misunderstanding something. usb.SearchFilter{} is ignored on linux, so wouldn't work anyway, the vendorID/productID function is the only one that matters there.

Reply:
When redoing this repo I needed to alter the searching function being used as search.Devices was no longer available. During development I turned the vendorID/productID filter off (see below that the function just returns true). @Otterverse should we re-add the filter for the PR?

Reply:
Vendor/Product should be enough to always get the right device and it'd be much better to use the device file returned from that rather than hardcoding a generic device name like /dev/ttyUSB0. The actual "SearchFilter" is mac only, and if anything may need to be expanded (not narrowed) to include additional device classes if this doesn't show up as a typical serial/dialup device under Mac.

Reply:
I don't actually understand this part - can you walk me through it so I can change this? E.g. a code-snippet or solution would work :)

Reply:
never mind - I implemented this using the search functionality and it seems to work well so far.

Reply:
Done.